app/app.go

```diff
@@ -556,11 +559,37 @@ func wirePrioritise(ctx context.Context, conf Config, life *lifecycle.Manager, t
 
 // wireRecaster wires the rebroadcaster component to scheduler, sigAgg and broadcaster.
 // This is not done in core.Wire since recaster isn't really part of the official core workflow (yet).
-func wireRecaster(sched core.Scheduler, sigAgg core.SigAgg, broadcaster core.Broadcaster) {
+func wireRecaster(ctx context.Context, sched core.Scheduler, sigAgg core.SigAgg, broadcaster core.Broadcaster, validators []state.Validator) error {
 	recaster := bcast.NewRecaster()
+
+	for _, val := range validators {
+		// Check if the current cluster state supports pre-generate validator registrations.
+		if len(val.BuilderRegistration.Signature) == 0 ||
+			len(val.BuilderRegistration.Message.FeeRecipient) == 0 ||
+			len(val.BuilderRegistration.Message.PubKey) == 0 {
+			break
+		}
+
+		pubkey, err := core.PubKeyFromBytes(val.PubKey)
+		if err != nil {
+			return errors.Wrap(err, "core pubkey from bytes")
+		}
+
+		signedData, err := core.NewVersionedSignedValidatorRegistration(builderRegistrationToETH2(val.BuilderRegistration))
+		if err != nil {
+			return errors.Wrap(err, "new versioned signed validator registration")
+		}
+
+		if err = recaster.StorePregen(ctx, core.NewBuilderRegistrationDuty(-1), pubkey, signedData); err != nil {
```

Comment:
lets put this in slot 0 and change the proper registration endpoint to use slot 1. Also hardcode network genesis timestamp in `eth2util` package and use genesis time for registration timestamp.

Can do this in a subsequent PR actually

Reply:
Okay so I have added timestamp in eth2util package for each network in this PR. This was needed for simnet tests.